File: csharp/adapter_practice/test/GoogleAdapterTest.cs
using adapter_practice.main.adapter;
using adapter_practice.main.document;
using adapter_practice.main.other;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Text;

namespace adapter_practice.test
{
    [TestFixture]
    class GoogleAdapterTest
    {
        /**
    * Google document.
    */
        private GoogleDoc gd;
        /**
         * Google adapter.
         */
        private DocsAdapter ga;

        /**
         * Set up method.
         */
        [SetUp]
    public void SetUp()
        {
            gd = new GoogleDoc();
            ga = new DocsAdapter(gd);
        }
        /**
         * Format test.
         */
        [TestCase]
    public void formatTest()
        {
            Object format = new Object();
            Assert.AreEqual(true, format.Equals(ga.getFormat()));
        }
        /**
         * Background test.
         */
        [TestCase]
        public void backgroundTest()
        {
            Image background = new Image();
            Assert.AreEqual(null, background.Equals(ga.getBackground()));
        }
        /**
         * Set office version test.
         */
        [TestCase]
        public void setVersionTest()
        {
            float officeVersion = 10;
            Assert.AreEqual("New office version is 10", ga.setMSOfficeVersion(officeVersion));
        }
        /**
         * Get license test.
         */
        [TestCase]
        public void getLicenseTest()
        {
            MSLicense license = new MSLicense();
            Assert.AreEqual(null, license.Equals(ga.getLicense()));
        }
        /**
         * restrict Edit test.
         */
        [TestCase]
    public void restrictTest()
        {
            MSLicense license = new MSLicense();
            Assert.AreEqual(false, ga.restrictEditIfLicenseIsInvalid(license));
        }
    }

}

File: java/src/test/java/mx/iteso/adapter/GoogleAdapterTest.java
package mx.iteso.adapter;

import mx.iteso.adapter.impl.other.Image;
import mx.iteso.adapter.impl.adapter.DocsAdapter;
import mx.iteso.adapter.impl.document.GoogleDoc;
import mx.iteso.adapter.impl.document.WordDoc;
import mx.iteso.adapter.impl.ints.IWordDocument;
import mx.iteso.adapter.impl.other.MSLicense;
import static org.junit.Assert.assertEquals;
import static org.junit.Test;
import static org.junit.Before;
/**
 * Word documents test class.
 */
public class GoogleAdapterTest {
    /**
     * Google document.
     */
    private GoogleDoc gd;
    /**
     * Google adapter.
     */
    private DocsAdapter ga;

    /**
     * Set up method.
     */
    @Before
    public void SetUp() {
        gd = new GoogleDoc();
        ga = new DocsAdapter(gd);
    }
    /**
     * Format test.
     */
    @Test
    public void formatTest() {
        Object format = new Object();
        asssertEquals(true, format.equals(ga.getFormat()));
    }
    /**
     * Background test.
     */
    @Test
    public void backgroundTest() {
        Image background = new Image();
        assertEquals(null, background.equals(ga.getBackground()));
    }
    /**
     * Set office version test.
     */
    @Test
    public void setVersionTest() {
        float officeVersion = 10;
        assertEquals("New office version is 10", ga.setMSOfficeVersion(officeVersion));
    }
    /**
     * Get license test.
     */
    @Test
    public void getLicenseTest() {
        MSLicense license = new MSLicense();
        assertEquals(null, license.equals(ga.getLicense()));
    }
    /**
     * restrict Edit test.
     */
    @Test
    public void restrictTest() {
        MSLicense license = new MSLicense();
        assertEquals(false, ga.restrictEditIfLicenseIsInvalid(license));
    }
}
